### PR TITLE
Source Facebook Marketing: Add lookback window to insights streams

### PR DESCRIPTION
**Pull request by vladimir-remar**
## What
Bring back the possibility to set up an attribution window. 
Let say the users will not use all the data retrieve from the insight stream and just use part of it (e.g., spend) in some cases 
set a `looback_window` short than 28 days would help the users by reducing the sync time.

## How
Set  value for  `insights_lookback_window` throught config.

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Comment by vladimir-remar** [comment]
@sherifnada I made this Draft related to [#9805](https://github.com/airbytehq/airbyte/pull/9805#issuecomment-1109858239)

**Comment by marcosmarxm** [comment]
@vladimir-remar this is ready to review?

**Comment by marcosmarxm** [comment]
@lazebnyi can you check the implementation @vladimir-remar is doing for Facebook Marketing?

**Comment by marcosmarxm** [comment]
@lazebnyi can you check the implementation @vladimir-remar is doing for Facebook Marketing?

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py
Comment (sherifnada) [inline airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py:151]: we should make this `default_insights_lookback_window` and apply it only to the default insights synced by this connector, and separately we should allow configuring this on a per-insight level in the custom insights

Comment (alafanechere) [reply airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py:168]: Hey @vladimir-remar I'm not sure you ended up implementing @sherifnada suggestion. Is anything blocking you from going in the suggested direction?

**Review by alafanechere — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py
Comment (vladimir-remar) [reply airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py:168]: Hello @alafanechere, maybe I'm wrong, I just replicate the same approach in the custom insight part.
https://github.com/airbytehq/airbyte/blob/c6d00d77a47a830f9bda3da6798734828de0e383/airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py#L80

**Review by alafanechere — CHANGES_REQUESTED** [review]
@vladimir-remar I think you're right, @sherifnada's concern is addressed, 👍 I'm requesting change as I think you need to update unit tests to cover your changes. This connector is in General Availability, meaning we are quite strict in terms of coverage. Could you please bump the connector version in the Dockerfile and update the changelog in `facebook-marketing.md`?

**Comment by alafanechere** [comment]
I would also need an additional review from @sherifnada or @lazebnyi  🙏

**Comment by lazebnyi** [comment]
@alafanechere I think @grubberr can help with review. Already discussed with him about that. He working now on FB marketing.

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py
Comment (grubberr) [inline airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py:86]: default value = 28
maximum value is 27 (because 28 is exclusive)

as for me it's strange ?

Comment (grubberr) [reply airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py:86]: Can you also provide here `Minimum` value to be more explicit ?

**Review by grubberr — CHANGES_REQUESTED** [review]

**Comment by grubberr** [comment]
@vladimir-remar please also fix unit_tests

```
pytest unit_tests


Results (1.28s):
      82 passed
      16 failed
```

**Comment by grubberr** [comment]
[SAT](https://docs.airbyte.com/connector-development/testing-connectors/source-acceptance-tests-reference) tests still has to be fixed https://github.com/airbytehq/airbyte/runs/6534873992?check_suite_focus=true#step:11:2631

**Comment by alafanechere** [comment]
@misteryeo could you please review this PR? It's changing the specs in `spec.py`

**Comment by vladimir-remar** [comment]
> Hey @grubberr do you have an idea why tests are failing now after passing yesterday? Did you merge master in the branch? @vladimir-remar I will be off until next Monday, bare with me for the review delay. @grubberr feel free to proceed to publish and merge once we get a review from @misteryeo .

@vladimir-remar Yes I did today after see the conflict.

**Comment by grubberr** [comment]
@vladimir-remar I think unit_tests failed because the master branch changed and you need to synchronise your code base with the latest updates in the master

**Comment by vladimir-remar** [comment]
> @vladimir-remar I think unit_tests failed because the master branch changed and you need to synchronise your code base with the latest updates in the master

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py
Comment (grubberr) [inline airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py:82]: do we need round brackets here ?

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py
Comment (grubberr) [inline airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py:170]: do we need round brackets here ?

File: airbyte-integrations/connectors/source-facebook-marketing/unit_tests/test_base_insight_streams.py
Comment (grubberr) [inline airbyte-integrations/connectors/source-facebook-marketing/unit_tests/test_base_insight_streams.py:330]: Can we remove this `monkeypatch` and pass `insights_lookback_window` as parameter to `AdsInsights` constructor like this:

```
stream = AdsInsights(api=api, start_date=start_date, end_date=end_date, insights_lookback_window=20)
```

?

**Review by grubberr — CHANGES_REQUESTED** [review]

**Review by grubberr — APPROVED** [review]

**Review by alafanechere — APPROVED** [review]
Thank you @grubberr for the review and @vladimir-remar for the changes! I'm now waiting for @misteryeo approval to publish and merge.

**Review by misteryeo — APPROVED** [review]

**Comment by alafanechere** [comment]
Thank you @vladimir-remar for the contribution! The new connector version is now published and I'm going to merge this PR.

**Pull request merged**
